ipyleaflet/leaflet.py

```diff
@@ -816,6 +822,21 @@ class Heatmap(RasterLayer):
     blur = Float(15.0).tag(sync=True, o=True)
     gradient = Dict({0.4: 'blue', 0.6: 'cyan', 0.7: 'lime', 0.8: 'yellow', 1.0: 'red'}).tag(sync=True, o=True)
 
+    def __init__(self, **kwargs):
+        super(Heatmap, self).__init__(**kwargs)
+        self.data = self._get_data()
+
+    @observe('gradient')
+    def _updata_data(self, change):
+        self.data = self._get_data()
+
+    def _get_data(self):
+        values = list(self.gradient.keys())
+        colors = list(self.gradient.values())
+        self.vmin = values[0]
+        self.vmax = values[len(values) - 1]
+        self.colormap = LinearColormap(colors, vmin=self.vmin, vmax=self.vmax)
```

Comment:
With the way `gradient` is defined, is it really a linear color map?

Reply:
Thanks for this question. There was an argument missing to properly set a piecewise linear colormap as shown on this picture for a gradient set to {0.1 :'orange', 0.2: 'red', 1.0: 'black'}
![piecewise_linear_colormap](https://user-images.githubusercontent.com/99649086/183877417-c9f6072f-0f89-445a-93ae-4eaffdd1471c.png)